lib/egon/overcloud/undercloud_handle/node.rb

```diff
@@ -85,7 +85,22 @@ def create_nodes_from_csv(csv_file)
     end
 
     def delete_node(node_id)
-      service('Baremetal').nodes.destroy(node_id)
+      begin
+        node = get_node(node_id)
+        if node.power_state != 'power off' && node.provision_state != 'active'
+          node.set_power_state('power off')
+          retries = 15
+          while retries > 0 && node.power_state != 'power off' do
+            sleep(2)
+            retries -= 1
+          end
+        end
+        service('Baremetal').nodes.destroy(node_id)
```

Comment:
So if the node is off, but the provision_state is 'active', it will still execute the destroy.  Is that what we want?